--- a/libcxx/CMakeLists.txt	2023-03-17 13:26:07.000000000 +0100
+++ b/libcxx/CMakeLists.txt	2023-05-01 21:07:07.083447258 +0200
@@ -763,14 +763,6 @@
     if (LIBCXX_BUILTINS_LIBRARY)
       target_link_libraries(${target} PRIVATE "${LIBCXX_BUILTINS_LIBRARY}")
     endif()
-  elseif (LIBCXX_HAS_GCC_LIB)
-    target_link_libraries(${target} PRIVATE gcc)
-  elseif (LIBCXX_HAS_GCC_S_LIB)
-    target_link_libraries(${target} PRIVATE gcc_s)
-  endif()
-
-  if (LIBCXX_HAS_ATOMIC_LIB)
-    target_link_libraries(${target} PRIVATE atomic)
   endif()
 
   if (MINGW)
